diff -Naru /nix/store/6djbrnvajq1zavalvv17aq33j87mlpgn-seastar-11ad0b1/core/memory.cc seastar-11ad0b1/core/memory.cc
--- /nix/store/6djbrnvajq1zavalvv17aq33j87mlpgn-seastar-11ad0b1/core/memory.cc	1970-01-01 08:00:01.000000000 +0800
+++ seastar-11ad0b1/core/memory.cc	2017-11-10 08:48:48.264580033 +0800
@@ -1348,7 +1348,10 @@
 
             if (r == -1) {
                 char err[1000] = {};
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
                 strerror_r(errno, err, sizeof(err));
+#pragma GCC diagnostic pop
                 std::cerr << "WARNING: unable to mbind shard memory; performance may suffer: "
                         << err << std::endl;
             }
diff -Naru /nix/store/6djbrnvajq1zavalvv17aq33j87mlpgn-seastar-11ad0b1/core/reactor.cc seastar-11ad0b1/core/reactor.cc
--- /nix/store/6djbrnvajq1zavalvv17aq33j87mlpgn-seastar-11ad0b1/core/reactor.cc	1970-01-01 08:00:01.000000000 +0800
+++ seastar-11ad0b1/core/reactor.cc	2017-11-10 08:47:17.116055798 +0800
@@ -2770,7 +2770,10 @@
     future<> loop_done = repeat([this] {
         return _aio_eventfd->readable().then([this] {
             char garbage[8];
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
             ::read(_aio_eventfd->get_fd(), garbage, 8); // totally uninteresting
+#pragma GCC diagnostic pop
             return _stopping ? stop_iteration::yes : stop_iteration::no;
         });
     });
@@ -2785,7 +2788,10 @@
         return;
     }
     uint64_t one = 1;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
     ::write(_aio_eventfd->get_fd(), &one, 8);
+#pragma GCC diagnostic pop
 }
 
 inline
